rasa/core/broker.py

```diff
@@ -56,21 +60,71 @@ def publish(self, event: Dict[Text, Any]) -> None:
         raise NotImplementedError("Event broker must implement the `publish` method.")
 
 
+def create_rabbitmq_ssl_options(
+    rabbitmq_host: Text = None
+) -> Optional[pika.SSLOptions]:
+    """Create RabbitMQ SSL options.
+
+    Requires the following environment variables to be set:
+
+    RABBITMQ_SSL_CLIENT_CERTIFICATE - path to the SSL client certificate (required)
+    RABBITMQ_SSL_CLIENT_KEY - path to the SSL client key (required)
+    RABBITMQ_SSL_CA_FILE - path to the SSL CA file for verification (optional)
+
+    Details on how to enable RabbitMQ TLS support can be found here:
+    https://www.rabbitmq.com/ssl.html#enabling-tls
+
+    Args:
+        rabbitmq_host: RabbitMQ hostname
+
+    Returns:
+        Pika SSL context of type `pika.SSLOptions` if all three paths are provided,
+        else `None`.
+
+    """
+
+    ca_file_path = os.environ.get("RABBITMQ_SSL_CA_FILE", None)
+    client_certificate_path = os.environ.get("RABBITMQ_SSL_CLIENT_CERTIFICATE", None)
+    client_key_path = os.environ.get("RABBITMQ_SSL_CLIENT_KEY", None)
+
+    if client_certificate_path and client_key_path:
+        logger.debug(
+            "Configuring SSL context for RabbitMQ host '{}'.".format(rabbitmq_host)
+        )
+
+        if ca_file_path:
+            logger.debug("Using certificate verification with provided CA file.")
+            context = ssl.create_default_context(cafile=ca_file_path)
+        else:
+            context = ssl.create_default_context(purpose=ssl.Purpose.CLIENT_AUTH)
+
+        context.load_cert_chain(client_certificate_path, client_key_path)
+
+        return pika.SSLOptions(context, rabbitmq_host)
+    else:
+        logger.debug(
+            "Cannot find SSL context for RabbitMQ host '{}'.".format(rabbitmq_host)
+        )
+
+        return None
+
+
 class PikaProducer(EventChannel):
     def __init__(
         self,
         host: Text,
         username: Optional[Text],
         password: Optional[Text],
+        port: Union[int, Text] = 5672,
         queue: Text = "rasa_core_events",
         loglevel: Text = logging.WARNING,
     ):
         import pika
```

Comment:
do we need to reimport that if we're globally importing now?

Reply:
no, will remove the global one again